server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1212,6 +1213,7 @@ public synchronized void compareTabletInfo(MetadataUpdateCount updateCounter,
       } else {
         log.error("Data files in {} differ from in-memory data {} {} {} {}", extent,
             tabletMetadata.getFilesMap(), dataFileSizes, updateCounter, latestCount);
+        getDatafileManager().handleMetadataDiff(context);
```

Comment:
Given #3721 , I'm not sure that the metadata table is guaranteed to be updated when this call completes.